containers-tests/tests/intmap-properties.hs

```diff
@@ -1187,3 +1542,41 @@ prop_fromSet :: [(Int, Int)] -> Bool
 prop_fromSet ys =
   let xs = List.nubBy ((==) `on` fst) ys
   in fromSet (\k -> fromJust $ List.lookup k xs) (IntSet.fromList $ List.map fst xs) == fromList xs
+
+newtype Identity a = Identity a
+    deriving (Eq, Show)
+
+instance Functor Identity where
+  fmap f (Identity a) = Identity (f a)
+
+instance Applicative Identity where
+  pure a = Identity a
+  Identity f <*> Identity a = Identity (f a)
+
+prop_traverseWithKey_identity :: IntMap A -> Property
+prop_traverseWithKey_identity mp = mp === newMap
+  where Identity newMap = traverseWithKey (\_ -> Identity) mp
+
+prop_traverseWithKey_degrade_to_mapWithKey :: Fun (Int, A) B -> IntMap A -> Property
+prop_traverseWithKey_degrade_to_mapWithKey fun mp =
+    mapWithKey f mp === newMap
+  where f = applyFun2 fun
+        g k v = Identity $ f k v
+        Identity newMap = traverseWithKey g mp
+
+prop_traverseMaybeWithKey_identity :: IntMap A -> Property
+prop_traverseMaybeWithKey_identity mp = mp === newMap
+  where Identity newMap = traverseMaybeWithKey (\_ -> Identity . Just) mp
+
+prop_traverseMaybeWithKey_degrade_to_mapMaybeWithKey :: Fun (Int, A) (Maybe B) -> IntMap A -> Property
+prop_traverseMaybeWithKey_degrade_to_mapMaybeWithKey fun mp =
+    mapMaybeWithKey f mp === newMap
+  where f = applyFun2 fun
+        g k v = Identity $ f k v
+        Identity newMap = traverseMaybeWithKey g mp
+
+prop_traverseMaybeWithKey_degrade_to_traverseWithKey :: Fun (Int, A) B -> IntMap A -> Property
+prop_traverseMaybeWithKey_degrade_to_traverseWithKey fun mp =
+    traverseWithKey f mp === traverseMaybeWithKey g mp
+  where f k v = Identity $ applyFun2 fun k v
+        g k v = fmap Just $ f k v
```

Comment:
The `Identity` functor is not convincing for this purpose. Could you pick something where the order of effects matters, etc.?

Reply:
Used `(String, a)` since that uses the `Monoid` class to produce the resulting `String`.